extra/markdown-extra/LeagueMarkdownFactory.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Twig\Extra\Markdown;
+
+use League\CommonMark\CommonMarkConverter;
+use League\CommonMark\Extension\ExtensionInterface;
+
+final class LeagueMarkdownFactory
+{
+    private $extensions;
+    private $converter;
+
+    /**
+     * @param ExtensionInterface[] $extensions
+     */
+    public function __construct(iterable $extensions, CommonMarkConverter $converter = null)
+    {
+        $this->extensions = $extensions;
+        $this->converter = $converter ?: new CommonMarkConverter();
```

Comment:
this does not make sense. Invoking the factory multiple times must not share the converter.